sporkfish/endgame_tablebase.py

```diff
@@ -110,9 +119,106 @@ def _should_query(self, board: Board, max_syzygy_size: int = 6) -> bool:
                 return False
         return True
 
+    def _categorize_dtz(self, dtz: int) -> "EndgameTablebase.DTZCategory":
+        """
+        Categorizes the Distance to Zeroing (DTZ) value into different categories.
+
+        :param dtz: The DTZ value to categorize, from our perspective.
+        :type dtz: int
+        :return: The category of the DTZ value.
+        :rtype: EndgameTablebase.DTZCategory
+        """
+        assert isinstance(
+            dtz, int
+        ), f"Expected DTZ to be of type int but got {type(dtz).__name__}."
+        if dtz < -100:
+            return EndgameTablebase.DTZCategory.CURSED_LOSS
+        elif -100 <= dtz <= -1:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_LOSS
+        elif dtz == 0:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_DRAW
+        elif 1 <= dtz <= 100:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_WIN
+        else:
```

Comment:
Sorry for being a pain - suggest being explicit here as well (i.e. using `elif` instead of `else`)

This is not as necessarily as the other case, since you have already enforced that `dtz` must be an int, so it is really just a style thing. I will let you decide if you agree with my suggestion.

Reply:
I'd rather keep it for now if ok - this is only called by query and we already have the assert to keep safe